catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePollerRunner.java

```diff
@@ -198,4 +186,76 @@ public void shutdown() {
         }
         LOGGER.trace("Status threads shut down");
     }
+
+    private SourceKey getSourceKey(Source source) {
+        return new SourceKey(source);
+    }
+
+    private static class SourceKey {
```

Comment:
It needs equals and hashcode. I will also add it to the CachedSource object because it is missing them.